daemon/endpoint.go

```diff
@@ -45,45 +44,44 @@ func NewGetEndpointHandler(d *Daemon) GetEndpointHandler {
 func (h *getEndpoint) Handle(params GetEndpointParams) middleware.Responder {
 	log.WithField(logfields.Params, logfields.Repr(params)).Debug("GET /endpoint request")
 
-	var wg sync.WaitGroup
+	modelChan := make(chan *models.Endpoint, 1)
+	defer close(modelChan)
+
+	// FIXME @aanm hide use of endpoint manager lock
+	endpointmanager.Mutex.RLock()
+	numEndpoints := len(endpointmanager.Endpoints)
+	eps := make([]*models.Endpoint, 0, numEndpoints)
 
 	if params.Labels == nil {
-		i := 0
-		endpointmanager.Mutex.RLock()
-		eps := make([]*models.Endpoint, len(endpointmanager.Endpoints))
-		wg.Add(len(endpointmanager.Endpoints))
-		for k := range endpointmanager.Endpoints {
-			go func(wg *sync.WaitGroup, i int, ep *endpoint.Endpoint) {
-				eps[i] = ep.GetModel()
-				wg.Done()
-			}(&wg, i, endpointmanager.Endpoints[k])
-			i++
+		for _, v := range endpointmanager.Endpoints {
+			go func(ep *endpoint.Endpoint) {
+				modelChan <- ep.GetModel()
+			}(v)
 		}
 		endpointmanager.Mutex.RUnlock()
-		wg.Wait()
+		for i := 0; i < numEndpoints; i++ {
+			eps = append(eps, <-modelChan)
```

Comment:
`eps[i] = <-modelChan`. You've allocated the slice on line 53.

Reply:
I like this but then `make()` on line 53 must be changed to pass `numEndpoints` as length and not just capacity.

Reply:
I'm merging this as-is so we can include it into the release today. I agree that we can beautify this.